test/e2e/storage/testsuites/snapshottable.go

```diff
@@ -139,23 +141,63 @@ func (s *snapshottableTestSuite) DefineTests(driver storageframework.TestDriver,
 			sc = volumeResource.Sc
 			claimSize = pvc.Spec.Resources.Requests.Storage().String()
 
-			ginkgo.By("starting a pod to use the claim")
+			ginkgo.By("[init] starting a pod to use the claim")
 			originalMntTestData = fmt.Sprintf("hello from %s namespace", pvc.GetNamespace())
 			command := fmt.Sprintf("echo '%s' > %s", originalMntTestData, datapath)
 
-			RunInPodWithVolume(cs, f.Timeouts, pvc.Namespace, pvc.Name, "pvc-snapshottable-tester", command, config.ClientNodeSelection)
+			pod := RunInPodWithVolume(cs, f.Timeouts, pvc.Namespace, pvc.Name, "pvc-snapshottable-tester", command, config.ClientNodeSelection)
 
 			err = e2epv.WaitForPersistentVolumeClaimPhase(v1.ClaimBound, cs, pvc.Namespace, pvc.Name, framework.Poll, f.Timeouts.ClaimProvision)
 			framework.ExpectNoError(err)
-			ginkgo.By("checking the claim")
+
 			// Get new copy of the claim
+			ginkgo.By("[init] checking the claim")
 			pvc, err = cs.CoreV1().PersistentVolumeClaims(pvc.Namespace).Get(context.TODO(), pvc.Name, metav1.GetOptions{})
 			framework.ExpectNoError(err)
 
 			// Get the bound PV
-			ginkgo.By("checking the PV")
-			_, err = cs.CoreV1().PersistentVolumes().Get(context.TODO(), pvc.Spec.VolumeName, metav1.GetOptions{})
+			ginkgo.By("[init] checking the PV")
+			pv, err := cs.CoreV1().PersistentVolumes().Get(context.TODO(), pvc.Spec.VolumeName, metav1.GetOptions{})
 			framework.ExpectNoError(err)
+
+			// At this point we know that:
+			// - a pod was created with a PV that's supposed to have data
+			// However there's a caching issue that @jinxu97 explained and it's related with the pod & volume
+			// lifecycle in windows, to understand it we first analyze what the volumemanager does:
+			// - when a container is no longer running the volumemanager will try to cleanup the volume mounts
+			//   - Unpublish: unbinds the global mount from the container
+			//     - Linux: the data is flushed to disk
+			//     - Windows: we delete a symlink, data's not flushed yet to disk
+			//   - Unstage: unmount the global mount
+			//     - Linux: disk is detached
+			//     - Windows: data is flushed to disk and the disk is detached
+			//
+			// Therefore the conclusion is that in Linux the data is already flushed to disk when the pod
+			// is deleted, however in Windows the data might not be flushed to disk yet when the pod is deleted
+			//
+			// For the next step in the test we have to take a snapshot however because of the problem
+			// described above we have to wait for a condition that signals us that the data is flushed to disk
+			// before taking the snapshot
+			//
+			// This is done by checking the actual state of the node, if it has attached volumes then
+			// it means that data might not have been flushed to disk yet
+			nodeName := pod.Spec.NodeName
```

Comment:
we could only trigger this logic for windows test.

Reply:
done